packages/wallet-connectors/src/WalletConnect.ts

```diff
@@ -401,22 +573,28 @@ export class WalletConnectConnector implements WalletConnector {
      * if the dApp doesn't have its own {@link https://cloud.walletconnect.com WalletConnect Cloud} project.
      * @param delegate The object to receive events emitted by the client.
      * @param network The network/chain that connected accounts must live on.
+     * @param connectionScope The scope of the connections, i.e. which methods and events to request permission for in the wallet.
+     * @param [modalConfig] The configuration of the modal for connecting to the mobile wallet. Defaults to the default invocation of {@linkcode createWalletConnectModalConfig}
      */
     static async create(
         signClientInitOpts: SignClientTypes.Options,
         delegate: WalletConnectionDelegate,
-        network: Network
+        network: Network,
+        connectionScope: WalletConnectConnectionScope,
+        modalConfig: WalletConnectModalConfig = createWalletConnectModalConfig(network)
     ) {
         const client = await SignClient.init(signClientInitOpts);
-        return new WalletConnectConnector(client, delegate, network);
+        return new WalletConnectConnector(client, delegate, network, connectionScope, modalConfig);
     }
 
     async connect() {
-        const { name } = this.network;
-
-        const chainId = `${WALLET_CONNECT_SESSION_NAMESPACE}:${name}`;
+        const chainId = getChainId(this.network);
```

Comment:
Consider storing `chainId` as a property on the class. Then you only need to resolve it from `create` and don't need the extracted function.